api/v1alpha1/release_types.go

```diff
@@ -138,128 +114,128 @@ type Release struct {
+	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// HasSucceeded checks whether the Release has succeeded or not.
-func (r *Release) HasSucceeded() bool {
-	return meta.IsStatusConditionTrue(r.Status.Conditions, releaseConditionType)
+// HasReleaseFinished returns a boolean indicating whether the Release's status indicates that it is done or not.
+func (r *Release) HasReleaseFinished() bool {
+	condition := meta.FindStatusCondition(r.Status.Conditions, releasedConditionType.String())
+	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// IsDeployed checks whether the Release has been successfully deployed via GitOps.
+// IsDeployed checks whether the Deployment has succeeded or not.
 func (r *Release) IsDeployed() bool {
-	condition := meta.FindStatusCondition(r.Status.Conditions, applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed)
-	return condition != nil && condition.Status == metav1.ConditionTrue
+	return meta.IsStatusConditionTrue(r.Status.Conditions, deployedConditionType.String())
 }
 
 // IsDeploying checks whether the Release has a valid start time for the deployment set in its status.
 func (r *Release) IsDeploying() bool {
-	return r.Status.DeploymentStartTime != nil && !r.Status.DeploymentStartTime.IsZero()
+	return r.Status.Deployment.StartTime != nil && !r.Status.Deployment.StartTime.IsZero()
 }
 
-// IsDone returns a boolean indicating whether the Release's status indicates that it is done or not.
-func (r *Release) IsDone() bool {
-	condition := meta.FindStatusCondition(r.Status.Conditions, releaseConditionType)
-	return condition != nil && condition.Status != metav1.ConditionUnknown
+// IsReleased checks whether the Release has succeeded or not.
+func (r *Release) IsReleased() bool {
+	return meta.IsStatusConditionTrue(r.Status.Conditions, releasedConditionType.String())
+}
+
+// IsReleasing checks whether the Release has a valid start time set in its status.
+func (r *Release) IsReleasing() bool {
+	return r.Status.Release.StartTime != nil && !r.Status.Release.StartTime.IsZero()
 }
 
 // MarkDeployed registers the deployment completion time and sets the AllComponentsDeployed status in the
 // Release to True with the provided reason and message.
-func (r *Release) MarkDeployed(reason, message string) {
-	if !r.IsDeploying() || (r.IsDeployed() && r.Status.DeploymentCompletionTime != nil) {
+func (r *Release) MarkDeployed() {
+	if !r.IsDeploying() || (r.HasDeploymentFinished() && r.Status.Deployment.CompletionTime != nil) {
 		return
 	}
 
-	r.Status.DeploymentCompletionTime = &metav1.Time{Time: time.Now()}
-	r.setStatusConditionWithMessage(applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed,
-		metav1.ConditionTrue, ReleaseReason(reason), message)
+	r.Status.Deployment.CompletionTime = &metav1.Time{Time: time.Now()}
+	conditions.SetCondition(&r.Status.Conditions, deployedConditionType, metav1.ConditionTrue, DeployedReason)
```

Comment:
The *whole* point of copying the status is to *not* have to go check the SEB. Why are we saying we are smarter than gitops and are adding our own logic layer to maintain (and for anyone looking at these CRs to understand)? Why can't we (continue to) say we are bubbling up the status from the SEB in the release? That's it. If someone says to us "the status in the release for deployment doesn't make much sense" we say that we are just bubbling up the status from the SEB, take it up with them. If the pipeline-service team had their own logic interpreting task results and put that on their pipelineruns instead of what the task actually reported, I would be extremely disappointed

Reply:
This is an old comment that was pending and sent with my last changes. What we discussed and agreed on offline (what we have implemented right now) is what prevails.